### PR TITLE
Tackle consumer starvation

### PR DESCRIPTION
**Pull request by glucaci**
It will be nice to control which background task is starting. In my case i want to separate the workflow poller from workflow consumer.

Setup:
- n x Workflow Producers (this can be achieved by not starting the host and just register the workflow)
- 1 x Workflow Poller (i want to be able to start the host without to have the consumer enabled)
- n x Workflow Consumers (i want to be able to start the host without to have the poller enabled)

The simple way is to add option for each background worker to can disable them and don't register it in DI container.

If you have another idea or thought about this please let me know.

Thanks

**Update:**
Short description of the problem: We are running the workflow host inside the application host and this is deployed in multiple instances. With  enough amount of runnable workflows and keeping the polling interval to 10s you bring consumers to starvations. That is why we wanted to have an option to deactivat the poller.

But know i found another solution that we can replace the `IGreyList` with a async ready variant and implement this in `MongoDB` or `Redis`

Let me know what you think.

### CODE REVIEW

File: src/WorkflowCore/Models/WorkflowOptions.cs [hunk 1]
```diff
@@ -34,6 +34,11 @@ public WorkflowOptions(IServiceCollection services)
             EventHubFactory = new Func<IServiceProvider, ILifeCycleEventHub>(sp => new SingleNodeEventHub(sp.GetService<ILoggerFactory>()));
         }
 
+        public bool DisableWorkflowConsumer { get; set; } = false;
+        public bool DisableEventConsumer { get; set; } = false;
+        public bool DisableIndexConsumer { get; set; } = false;
+        public bool DisableRunnablePoller { get; set; } = false;
```
Comment (danielgerlag) [inline src/WorkflowCore/Models/WorkflowOptions.cs:40]: What do you think of flipping the naming here and default to true... something like `ProcessWorkflows`, `ProcessEvents`, `PollingEnabled` or some variant?

**Review by danielgerlag — APPROVED** [review]

File: src/WorkflowCore/Models/WorkflowOptions.cs (hunk 1 above)
Comment (glucaci) [reply src/WorkflowCore/Models/WorkflowOptions.cs:40]: Yes, no problem

**Comment by glucaci** [comment]
Done. I'll do another PR for `IGreyList`

**Comment by danielgerlag** [comment]
@glucaci thanks... looks like the build is failing though

**Comment by glucaci** [comment]
@danielgerlag done

**Pull request merged**